Sources/MapboxCoreNavigation/PassiveLocationDataSource.swift

```diff
@@ -29,6 +29,7 @@ open class PassiveLocationDataSource: NSObject {
         super.init()
         
         self.systemLocationManager.delegate = self
+        navigator.addObserver(for: self)
```

Comment:
This implementation makes both PassiveLocationDataSource and RouteController individually register as observers, which leads to the questions about observation lifetime that you raised above. Since `MapboxCoreNavigation.Navigator` is already mirroring `MapboxNavigationNative.Navigator` to some extent, can we have the former register as an observer for the latter and broadcast updates to PassiveLocationDataSource and RouteController via notifications, similar to the existing progress change notifications?

Reply:
I don't see how this can solve the lifetime problem, RouteController anyway will have to subscribe to the updates.